src/main/java/org/onebusaway/alexa/AuthedSpeechlet.java

```diff
@@ -0,0 +1,170 @@
+        this.userData = userData;
+        obaUserClient = new ObaUserClient(
+                obaAgencies.agencyForCity(userData.getCity()).get());
+    }
+
+    @Override
+    public SpeechletResponse onIntent(final IntentRequest request,
+                                      final Session session)
+                                      throws SpeechletException {
+        Intent intent = request.getIntent();
+        if ("SetCityIntent".equals(intent.getName())) {
+            return anonSpeechlet.onIntent(request, session);
+        } else if ("GetCityIntent".equals(intent.getName())) {
+            PlainTextOutputSpeech out = new PlainTextOutputSpeech();
+            out.setText(
+                    String.format("You live in %s.",
+                            userData.getCity()));
+            return SpeechletResponse.newTellResponse(out);
+        } else if ("SetStopNumberIntent".equals(intent.getName())) {
+            return anonSpeechlet.onIntent(request, session);
+        } else if ("GetArrivalsIntent".equals(intent.getName())) {
+            return tellArrivals();
+        } else {
+            throw new SpeechletException("Did not recognize intent name");
+        }
+    }
+
+    @Override
+    public SpeechletResponse onLaunch(final LaunchRequest request,
+                                      final Session session)
+                                      throws SpeechletException {
+        return tellArrivals();
+    }
+
+    @Override
+    public void onSessionStarted(final SessionStartedRequest request,
+                                 final Session session) {
+
+    }
+
+    @Override
+    public void onSessionEnded(final SessionEndedRequest request,
+                               final Session session) {
+
+    }
+
+    private SpeechletResponse tellArrivals() {
+        ObaArrivalInfo[] arrivals = obaUserClient.getArrivalsAndDeparturesForStop(userData.getStopId());
+        StringBuilder sb = new StringBuilder();
+        for (ObaArrivalInfo arrival: arrivals) {
+            log.info("Arrival: " + arrival);
+            sb.append(String.format("Route %s, %s, %s. -- ",
+                    arrival.getShortName(),
+                    arrival.getHeadsign(),
+                    friendlyArrivalText(arrival)
+                    ));
+        }
+        log.info("Full text output: " + sb.toString());
+        PlainTextOutputSpeech out = new PlainTextOutputSpeech();
+        out.setText(sb.toString());
+        return SpeechletResponse.newTellResponse(out);
+    }
+
+    private String friendlyArrivalText(ObaArrivalInfo arrival) {
+        return (arrival.getPredicted() ?
+                friendlyPredictedArrivalText(arrival) :
+                friendlyScheduledArrivalText(arrival));
+    }
+
+    private String friendlyPredictedArrivalText(ObaArrivalInfo arrival) {
+        ZonedDateTime predictedArrivalTime = predictedArrivalOf(arrival).get();
+        long mins_to_arrival = ChronoUnit.MINUTES.between(
```

Comment:
`arrival.getCurrentTime()` should always return the current time in UTC of the OBA server.  My original thought was to always compare this to the predicted arrival/departure times in UTC when producing ETA to avoid any clock sync issues.  I'm fine with this as-is, although we should keep an eye out for clock sync problems.  I'm also used to working on mobile where device clocks can vary - I assume AWS is somewhat more reliable :).